v2/controllers/egress_watcher.go

```diff
@@ -55,7 +55,25 @@ func (r *EgressWatcher) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		return ctrl.Result{}, err
 	}
 
+	targetPods := make(map[string]corev1.Pod)
```

Comment:
It makes a large CPU footprint to store large map-values **by-value**, so I suspect it is generally better to store map-values **by-reference** unless the values are to be modified.

```suggestion
	targetPods := make(map[string]*corev1.Pod)
```

Reply:
In fact, I guess we don't need to have pods as values.
Can't we use `map[string]struct{}`?

Reply:
We use the filtered pods as reconciliation targets.

https://github.com/cybozu-go/coil/pull/280/files#diff-3eca2136d6390132658b46160f9657817add56a34fb13fc573544e0aadbe49c5R76

Reply:
@terassyi 
It seems like we can remove the second `for` loop.

Reply:
Before processing a reconciliation, I want to check whether some pods have the same address.
And if there is a duplication, I want to return the error immediately.
Therefore, I filter all pods in advance.

Reply:
fair enough